p2p/host/autonat/autonat.go

```diff
@@ -199,16 +181,15 @@ func (as *AmbientAutoNAT) background() {
 		case e := <-subChan:
 			switch e := e.(type) {
 			case event.EvtLocalAddressesUpdated:
-				if !lastAddrUpdated.Add(time.Second).After(time.Now()) {
-					lastAddrUpdated = time.Now()
-					if as.confidence > 1 {
-						as.confidence--
-					}
+				// On local address update, reduce confidence from maximum so that we schedule
```

Comment:
What is a local address update?

Reply:
this is any change in the host address list(host.AllAddrs()). I used the phrase from the event EvtLocalAddressesUpdated.

I didn't change this much since it was not related to this PR, but the correct thing to do here would be to only reduce confidence if there are any addresses removed from the host address list and our status is public or if there are addresses added and our status is private.

Reply:
Yeah, this is fine. Sorry, I messed this up, didn't realize this was literally in the name of the event ;)